src/claimable_balances/all_claimable_balances_request.rs

```diff
@@ -28,71 +29,9 @@ pub struct AllClaimableBalancesRequest {
     order: Option<Order>,
 }
 
-impl Request for AllClaimableBalancesRequest {
-    fn new() -> Self {
-        AllClaimableBalancesRequest {
-            sponsor: None,
-            asset: None,
-            claimant: None,
-            cursor: None,
-            limit: None,
-            order: None,
-        }
-    }
-
-    fn get_path(&self) -> &str {
-        "/claimable_balances/"
-    }
-
-    fn get_query_parameters(&self) -> String {
-        let mut query = String::new();
-        if let Some(sponsor) = &self.sponsor {
-            query.push_str(&format!("sponsor={}&", sponsor));
-        }
-        if let Some(asset) = &self.asset {
-            query.push_str(&format!("asset={}&", asset));
-        }
-        if let Some(claimant) = &self.claimant {
-            query.push_str(&format!("claimant={}&", claimant));
-        }
-        if let Some(cursor) = &self.cursor {
-            query.push_str(&format!("cursor={}&", cursor));
-        }
-        if let Some(limit) = &self.limit {
-            query.push_str(&format!("limit={}&", limit));
-        }
-        if let Some(order) = &self.order {
-            query.push_str(&format!("order={}&", order));
-        }
-
-        query.trim_end_matches('&').to_string()
-    }
-
-    fn build_url(&self, base_url: &str) -> String {
-        format!(
-            "{}{}?{}",
-            base_url,
-            self.get_path(),
-            self.get_query_parameters()
-        )
-    }
-
-    fn validate(&self) -> Result<(), String> {
-        if let Some(sponsor) = &self.sponsor {
-            let is_valid = is_public_key(sponsor);
-            if is_valid.is_err() {
-                return Err(is_valid.unwrap_err());
-            }
-        }
-
-        if let Some(claimant) = &self.claimant {
-            let is_valid = is_public_key(claimant);
-            if is_valid.is_err() {
-                return Err(is_valid.unwrap_err());
-            }
-        }
-
-        Ok(())
+impl AllClaimableBalancesRequest {
```

Comment:
A lot of the comments from the all assets request also apply here. There should be an `InvalidClaimableBalancesRequest` state, which does not allow for an empty request without any parameters.
Don't forget to add boundaries to the documentation, and check the type for limit, to trim it down to the smallest possible value that matches the limits of the request.